ci(e2e): add DVP-over-DVP matrix testing with parallel execution

yachmenevas opened the pull request:
## Description
<!---
  Describe your changes with technical details.
-->
Implement isolated DVP-over-DVP testing with parallel matrix execution for storage profiles (sds, cephrbd). Includes modular workflow architecture, CLI wrapper, and automatic cleanup for nightly runs.

## Why do we need it, and what problem does it solve?


## What is the expected result?
<!---
  Describe steps to reproduce the expected result.
  What ACTION(s) to take to ensure the problem is gone.
-->


## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.


## Changelog entries
<!---
  /!\ See CONTRIBUTING.md for more details. /!\
  Examples:
  ```changes
  section: core
  type: feature
  summary: "Node restarts can be avoided by pinning a checksum to a node group in config values."
  ---
  section: core
  type: fix
  summary: "Nodes with outdated manifests are no longer provisioned on *InstanceClass update."
  impact_level: high
  impact: |
    Expect nodes of "Cloud" type to restart.
  ---
  impact_level: low
  ```
-->

```changes
section: ci
type: chore
summary: add DVP-over-DVP matrix e2e testing
```

yachmenevas committed:
ci(e2e): add DVP-over-DVP matrix testing with parallel execution

Signed-off-by: Yachmenev Anton <anton.yachmenev@flant.com>

yachmenevas committed:
ci: drop artifacts/binaries from VCS; update .gitignore

Signed-off-by: Yachmenev Anton <anton.yachmenev@flant.com>

yachmenevas committed:
ci(e2e): restore workflow, Taskfiles, charts, scripts and docs

Signed-off-by: Yachmenev Anton <anton.yachmenev@flant.com>

yachmenevas committed:
dvcr: make templates nil-safe; remove temporary root values.yaml

Signed-off-by: Yachmenev Anton <anton.yachmenev@flant.com>

yachmenevas committed:
ci: format YAML to satisfy prettier; drop temporary values.yaml

yachmenevas committed:
revert(templates/dvcr): drop E2E-related edits; keep tests via ModuleConfig/Taskfile

Signed-off-by: Yachmenev Anton <anton.yachmenev@flant.com>

yachmenevas committed:
ci(e2e): remove secrets usage in if; check webhook in step script

yachmenevas committed:
ci(e2e): write kubeconfig via printf to avoid heredoc EOF issues

yachmenevas committed:
ci(e2e): resolve merge with main; keep full E2E workflow

yachmenevas committed:
ci(e2e): fix duplicate workflow_dispatch; prettier passes

yachmenevas committed:
ci(e2e): use env.E2E_K8S_URL in kubeconfig steps; drop report kubeconfig env

yachmenevas committed:
ci(e2e): build e2e-runner wrapper via Taskfile (tests/e2e task run:ci)

yachmenevas committed:
ci(e2e): run tests via tests/e2e Taskfile (run:ci) with TIMEOUT

yachmenevas committed:
ci(e2e): pass TEST_TIMEOUT and E2E_PREFIX to tests/e2e Taskfile run

yachmenevas committed:
ci(e2e): pass TIMEOUT (not TEST_TIMEOUT) to tests/e2e Taskfile

yachmenevas committed:
ci(e2e): write kubeconfig token without base64 decoding to avoid UTF-8 issues

yachmenevas committed:
ci(e2e): quote token in kubeconfig; set LABELS/FOCUS_JSON defaults for jq

yachmenevas committed:
ci(e2e): read JWT from secrets.K8S_CLUSTER_SECRET index 8 (kubeconfig)

yachmenevas committed:
ci(e2e): use azure/k8s-set-context@v4 for kubeconfig setup

yachmenevas committed:
ci(e2e): remove KUBECONFIG env var to use default ~/.kube/config

yachmenevas committed:
ci(e2e): implement complete artifact collection and Loop reporting pipeline

- Add JUnit collection step after task run:ci
- Add matrix log collection with proper format
- Fix artifact uploads with correct paths (ci/dvp-e2e/artifacts and ci/dvp-e2e/tmp/matrix-logs)
- Fix artifact downloads in report job with merge-multiple
- Add debug step to troubleshoot artifact layout
- Fix kubeconfig handling in cleanup job

yachmenevas committed:
feat(ci): replace simple E2E tests with full DVP-over-DVP bootstrap

- Replace 'Run E2E tests (Taskfile)' with 'Run DVP-over-DVP E2E tests'
- Use nested:test-run task which includes:
  * Full DVP-over-DVP cluster bootstrap (infra + dhctl)
  * Storage profile configuration (sds/cephrbd)
  * E2E test execution against nested cluster
  * Automatic cleanup via defer
- Pass REGISTRY_DOCKER_CFG secret for Docker registry access
- Update JUnit collection logic for new file location
- Tests now run against fresh nested clusters instead of existing cluster

yachmenevas committed:
fix(ci): enable cancel-in-progress for concurrency

- Change cancel-in-progress from false to true
- This will cancel previous runs when new ones start
- Fixes pending state issue where jobs wait for themselves

yachmenevas committed:
ci(e2e): restore ci/dvp-e2e/Taskfile.vm.yaml (VM helper tasks)

yachmenevas committed:
ci(e2e): use existing kubeconfig from GA; skip ingress/RBAC during ensure

- Pass SKIP_INGRESS_CHECK=true to parent:kubeconfig:ensure-scoped
- Export SKIP_INGRESS_CHECK in workflow step
- Avoid RBAC operations (CRB) in parent cluster; fixes Forbidden error

yachmenevas committed:
ci(e2e): simplify kubeconfig provisioning; use E2E_NESTED_SA_SECRET only\n\n- Drop azure/k8s-set-context step\n- Build kubeconfig directly from secret JWT\n- Keep tests launch minimal

yachmenevas committed:
ci(e2e): tidy headings and finalize simplified flow (no validate, docker login, kubeconfig from secret, cleanup after report)

yachmenevas committed:
ci(e2e): pass registryDockerCfg via YAML overlay and use it in nested:test-run

yachmenevas committed:
ci(e2e): ensure REGISTRY_DOCKER_CFG overlay is present and non-empty

yachmenevas committed:
ci(e2e): merge REGISTRY_DOCKER_CFG into base values via yq; pass as VALUES_TEMPLATE_FILE

yachmenevas committed:
ci(e2e): inject REGISTRY_DOCKER_CFG into generated values for InitConfiguration

yachmenevas committed:
ci(e2e): map SCs per profile (sds→linstor-thin-r2; cephrbd→ceph) for nested

yachmenevas committed:
ci(e2e): fix registry auth and SC mapping in matrix workflow

yachmenevas committed:
ci(e2e): configure registry auth via REGISTRY_DOCKER_CFG in matrix

yachmenevas committed:
ci(e2e): wait for base VirtualImage Ready to prevent bootstrap hangs

yachmenevas committed:
ci(e2e): propagate TARGET_STORAGE_CLASS to infra; wait VirtualImage Ready; robust docker pull

yachmenevas committed:
revert: rollback recent infra SC and bootstrap waits to previous state

yachmenevas committed:
ci(e2e): ensure default StorageClass per profile before E2E run

yachmenevas committed:
ci(e2e): set infra storageClass from TARGET_STORAGE_CLASS to avoid parent default SC

yachmenevas committed:
ci(e2e): remove parent default SC step; rely on explicit infra storageClass

yachmenevas committed:
ci(e2e): cleanup uses privileged SA context; remove kubeconfig copy and soft fallback

yachmenevas committed:
ci(e2e): correct SDS image/snapshot SC mapping (IMG_SC=linstor-thin-r1-immediate)

yachmenevas committed:
ci(e2e): add -vv --github-output to ginkgo; use dvp-e2e- prefix for RUN_ID, junit, logs

yachmenevas committed:
ci(e2e): fix doubled namespace issue; cleanup filters dvp-e2e-; revert prefixed RUN_ID and artifact paths

yachmenevas committed:
ci(e2e): prune debug/local tasks; keep only CI-needed targets; purge artifacts after upload

yachmenevas committed:
ci(e2e): add JUnit failure summary step and cleanup filter dvp-e2e-nightly-* (debug)

yachmenevas committed:
ci(e2e): fix YAML inline python indentation for JUnit summary step

yachmenevas committed:
ci(e2e): switch JUnit summary to python -c to fix YAML parser error

yachmenevas committed:
ci(e2e): switch to 'go tool ginkgo' (was go run) for test runner

yachmenevas committed:
ci(e2e): avoid YAML parse issues by removing emojis from echo lines

yachmenevas committed:
ci(e2e): wait for validating-webhook-handler and retry ModuleConfig applies for SDS/Ceph

yachmenevas committed:
ci(e2e): set defaultClusterStorageClass in nested before running tests

yachmenevas committed:
ci(e2e): align SDS IMAGE_STORAGE_CLASS to linstor-thin-r2; use ginkgo via go run -v

yachmenevas committed:
ci(e2e): align with reference - ModuleConfig version 1, SDS StorageClass linstor-thin-r2

yachmenevas committed:
ci(e2e): fix kubeconfig logic - remove SKIP_INGRESS_CHECK, simplify SOURCE_KUBECONFIG

yachmenevas committed:
ci(e2e): ensure tools install on cache-hit; bump cache key to v3

yachmenevas committed:
ci(e2e): force kubectl into ~/.local/bin, ensure d8 in PATH; add detailed nested:e2e diagnostics

yachmenevas committed:
ci(e2e): align nested API warm-up with reference (20s readyz, 10x nodes with fail)

yachmenevas committed:
ci(e2e): align Ceph SC to ceph-pool-r2-csi-rbd-immediate and provisioner to rbd.csi.ceph.com; fix workflow mapping

Isteb4k commented on .github/actions/setup-e2e-tools/action.yml:
```diff
@@ -0,0 +1,80 @@
+name: Setup E2E Tools
+description: Install kubectl, helm, d8, task, yq with caching
```
No need to create your own action to setup kubectl, helm and so on. It already exists. Find it, please

Isteb4k commented on .github/workflows/e2e-matrix.yml:
```diff
@@ -12,23 +12,350 @@
-  noop:
-    name: Bootstrap
+  # ============================================
+  # 1. SETUP - Environment preparation
```
Describe in this comment what exactly you prepare here

Isteb4k commented on .github/workflows/e2e-matrix.yml:
```diff
@@ -12,23 +12,350 @@
+      - name: Load storage profiles
+        id: load
+        run: |
+          # For now, use hardcoded profiles. Later this can be made dynamic
```
Make it dynamic now

Isteb4k commented on .github/workflows/e2e-matrix.yml:
```diff
@@ -12,23 +12,350 @@
+        run: |
+          set -euo pipefail
+          mkdir -p ~/.kube
+          TOK='${{ secrets.E2E_NESTED_SA_SECRET }}'
```
Rename TOK to E2E_NESTED_SA_SECRET

Isteb4k commented on .github/workflows/e2e-matrix.yml:
```diff
@@ -12,23 +12,350 @@
+          chmod 600 ~/.kube/config
+          echo "KUBECONFIG=$HOME/.kube/config" >> "$GITHUB_ENV"
+
+      - name: Configure registry auth
```
Why do you need this?

Isteb4k commented on ci/dvp-e2e/Taskfile.yaml:
```diff
@@ -0,0 +1,2174 @@
+        fi
+      - |
+
+        export WK_SC='{{ .TARGET_STORAGE_CLASS }}'
```
What does WK mean?

universal-itengineer commented on ci/dvp-e2e/charts/infra/values.yaml:
```diff
@@ -0,0 +1,58 @@
+    workers:
+      root: sds-local-storage
+    infra:
+      nfs: nfs-4-1-wffc
```
Do we really need this?

universal-itengineer commented on ci/dvp-e2e/charts/infra/templates/nfs-server/deploy.yaml:
Remove this dir please.

universal-itengineer commented on .github/workflows/e2e-matrix.yml:
```diff
@@ -12,23 +12,350 @@
+                TEST_TIMEOUT="${{ inputs.timeout || '4h' }}"
+            fi
+          done
+
```
Cluster have only one storage in this tests case, you don't need to send individual reports for each profile

universal-itengineer commented on ci/dvp-e2e/scripts/ceph-bootstrap.sh:
To deploy the ceph, you need to use the deckhouse ModuleConfig

Commit:
refactor(ci): remove unused setup-e2e-tools action and improve E2E workflow

- Remove custom setup-e2e-tools action (replaced with standard GitHub Actions)
- Add dynamic profile configuration via profiles.json and get_profile_config.sh
- Separate monolithic E2E job into focused steps for better debugging
- Optimize Loop reporting (single matrix summary instead of individual reports)
- Extract Loop tasks to separate tasks/loop.yaml for modularity
- Fix finalizer removal practice (replace undeploy-finalizers.sh with kubectl wait)
- Remove unused Taskfile.vm.yaml from workflow
- Improve resource naming: nightly-nested-e2e-{profile}-{date} format
- Add comments to jump-host port 2222 usage
- Use azure/k8s-set-context@v4 and deckhouse/modules-actions/setup@v2

Commit:
fix(ci): remove extra profiles from profiles.json

- Keep only original profiles: sds and cephrbd
- Remove nfs and hostpath profiles that were added by mistake
- Restore original 2-profile matrix configuration

Commit:
refactor(ci): simplify cleanup and fix env variable issues

- Simplify cleanup:namespaces task (remove CONFIRM, comments, extra tasks)
- Fix GitHub Actions env variable handling in Configure storage profile step
- Remove problematic echo statements that caused env command errors
- Use proper  syntax for variable export
- Clean up unused cleanup tasks (artifacts, logs, complete)

Commit:
fix(ci): improve profile config parsing to avoid env errors

- Add ^ prefix to grep patterns for exact line matching
- Prevent parsing of stray lines without STORAGE_CLASS prefix
- Fix 'Invalid format' errors caused by loose grep matching
- Add comments for better code clarity

Commit:
fix(ci): use correct ServiceAccount and improve cleanup error handling

- Replace E2E_VIRTUALIZATION_SA_SECRET with E2E_NESTED_SA_SECRET
- Fix ServiceAccount permissions for ClusterRoleBinding creation
- Improve cleanup:namespaces task with better error handling
- Add namespace existence check before waiting for deletion
- Prevent cleanup failures when namespaces don't exist
- Add informative messages for cleanup progress

Commit:
ci(e2e): build kubeconfig from E2E_NESTED_SA_SECRET without TOK var; drop k8s-set-context

Commit:
ci(e2e): call correct cleanup task (cleanup:cleanup:namespaces)

Commit:
ci(e2e): drop NFS server from infra and tasks; gate NFS config by flag

Commit:
ci(e2e): docker login dev-registry before pulling install image

Commit:
ci(e2e): use REGISTRY_DOCKER_CFG for registry auth; drop explicit docker login

Commit:
ci(e2e): use deckhouse/modules-actions/setup for registry auth only

Commit:
ci(e2e): auth via modules-actions; add GHCR login; setup-task with token

Commit:
ci(e2e): add explicit registry pull check before bootstrap

Commit:
ci(e2e): replace Taskfile with minimal GH-oriented version; drop unused tasks/* and loop notify scripts; keep JSON profiles + modules-actions/ghcr auth; Ceph/SDS only

Commit:
ci(e2e): fix cleanup task; docker login to DEV_REGISTRY; setup-task input; lint YAML

Commit:
ci(e2e): remove GHCR login, move registry verification after docker login

Commit:
ci(e2e): use direct docker login instead of docker/login-action

Commit:
ci(e2e): use CE installer image instead of dev-registry for verification

Commit:
ci(e2e): use CE installer image in dhctl-bootstrap task

Commit:
ci(e2e): fix cleanup namespaces array init and reorder workflow steps for SSH key generation

Commit:
ci(e2e): fix workflow architecture - add proper SSH key generation and render-infra step

Commit:
ci(e2e): rely on render-infra ssh-gen dependency; remove explicit ssh:generate step

Commit:
ci(e2e): run values prepare before infra; remove intermediate base values creation

Commit:
fix(ci): align Taskfile dependencies with backup branch logic

Commit:
fix(ci): add sshPublicKey to cluster-config generation

Commit:
fix(ci): ensure deckhouse.registryDockerCfg is always a string in cluster-config

Commit:
fix(ci): remove quote from registryDockerCfg in cluster-config template

Commit:
fix(ci): remove registryDockerCfg field from cluster-config template

Commit:
fix(ci): restore registryDockerCfg config and fix cleanup namespaces

- Add registry auth step with REGISTRY_DOCKER_CFG merge into values
- Restore registryDockerCfg field in cluster-config template
- Add safe cleanup task with proper array initialization
- Switch cleanup workflow to use cleanup:namespaces:safe with CONFIRM=true

Commit:
fix(ci): use VALUES_TEMPLATE_FILE instead of explicit VALUES_FILE in dhctl-bootstrap

Commit:
fix(ci): use run:values:prepare to create proper values.yaml with dynamic namespace

- Replace manual values merge with existing run:values:prepare logic
- Add registryDockerCfg to the prepared values.yaml file
- Remove duplicate Prepare run values.yaml step
- Ensure proper namespace is used throughout the workflow

Commit:
fix(ci): use VALUES_TEMPLATE_FILE consistently and remove undeploy-finalizers script call

- Remove VALUES_FILE parameters from render-infra and infra-deploy tasks
- Let tasks use VALUES_TEMPLATE_FILE from env variable consistently
- Remove call to non-existent undeploy-finalizers.sh script from cleanup
- This ensures all tasks use the correct values.yaml with dynamic namespace

Commit:
debug(ci): add debugging info for REGISTRY_DOCKER_CFG secret

- Add debug output to check if REGISTRY_DOCKER_CFG secret is available
- Show secret length and available SECRET env vars for troubleshooting
- Add confirmation message when registryDockerCfg is added to values.yaml

Commit:
fix(ci): properly pass REGISTRY_DOCKER_CFG secret to workflow

- Add env section to Configure registry auth step
- Remove debug output and simplify logic
- Restore registryDockerCfg field with quote in template
- Remove conditional REGISTRY_DOCKER_CFG logic from Taskfile

Commit:
fix(ci): pass VALUES_FILE to dhctl-bootstrap task

- Add VALUES_FILE parameter to ensure run-specific values.yaml is used
- This fixes registryDockerCfg being null during bootstrap

Commit:
fix(ci): use dynamic namespace consistently across all steps

- Remove default namespace from values.yaml to prevent nightly-e2e creation
- Add VALUES_FILE to render-infra and infra-deploy steps
- Ensure all components use the same dynamic namespace

Commit:
cleanup: remove accidentally added submodules and docs

- Remove ci/dvp-over-dvp submodule
- Remove ci/local/dvp-over-dvp submodule
- Remove ci/virtualization-rnd directory
- Remove docs/internal/ci-e2e.md and ci-e2e.ru.md

Commit:
fix(ci): synchronize registry for CE and fix submodule error

- Update imagesRepo to CE registry in cluster-config.yaml
- Update dhctl-bootstrap image to CE stable in Taskfile.yaml
- Remove dangling submodule entry for ci/dvp-over-dvp from .gitmodules and .git/config
- Remove ci/dvp-over-dvp directory

Commit:
fix(ci): remove registryDockerCfg for CE registry

- Remove registryDockerCfg from cluster-config template (CE doesn't need creds)
- Remove REGISTRY_DOCKER_CFG step from workflow
- Keep run:values:prepare step for dynamic namespace creation
- This fixes 'cannot unmarshal docker cfg: unexpected end of JSON input' error

Isteb4k commented on ci/dvp-e2e/charts/cluster-config/templates/virtualization.yaml:
```diff
@@ -0,0 +1,25 @@
+metadata:
+  name: virtualization
+spec:
+  imageTag: {{ .Values.virtualization.tag }}
```
Why `.Values.virtualization.tag` not just `main`

Commit:
fix(ci): correct registry configuration for OSS cluster with CE dhctl

- Remove unused cluster settings from cluster-config/values.yaml
- Change imagesRepo back to OSS registry (dev-registry.deckhouse.io/sys/deckhouse-oss)
- Add registryDockerCfg back to cluster-config template for OSS auth
- Add REGISTRY_DOCKER_CFG step back to workflow
- Keep CE dhctl bootstrap image (registry.deckhouse.ru/deckhouse/ce/install:stable)

Architecture: CE dhctl creates OSS cluster with OSS registry and OSS credentials

Commit:
fix(ci): disable upmeter module and fix virtualization tag

- Add explicit upmeter ModuleConfig with enabled: false to mc.yaml
- Change virtualization imageTag from .Values.virtualization.tag to main
- This should resolve the upmeter timeout issue during bootstrap

Commit:
fix(ci): remove duplicate upmeter ModuleConfig

- Remove upmeter from disabled-modules.yaml to avoid duplication
- Keep only explicit upmeter ModuleConfig in mc.yaml
- This should resolve the upmeter timeout issue caused by duplicate ModuleConfigs

Commit:
fix(ci): consolidate disabled modules and remove submodule

- Move all disabled modules from disabled-modules.yaml to mc.yaml
- Remove disabled-modules.yaml to avoid conflicts
- Remove ci/virtualization-rnd submodule from git index
- This should resolve ModuleConfig conflicts and git submodule warnings

Commit:
fix(ci): completely remove virtualization-rnd directory

- Remove ci/virtualization-rnd directory completely
- This should resolve all git submodule warnings

Commit:
fix(ci): disable checkout caching in GitHub Actions

- Add fetch-depth: 0 to all actions/checkout@v4 steps
- This ensures fresh code is used instead of cached versions
- Should resolve issues with stale ModuleConfig causing upmeter timeouts

Commit:
fix(ci): skip checksum verification in Deckhouse CLI installation

- Add --skip-checksum flag to d8-install.sh
- This resolves sha256sum verification failures
- Allows workflow to proceed past CLI installation step

Commit:
fix(ci): restore disabled-modules.yaml and fix duplication

- Restore disabled-modules.yaml with local-path-provisioner, pod-reloader, secret-copier, namespace-configurator
- Keep upmeter only in mc.yaml to avoid duplication
- This should resolve ModuleConfig conflicts while maintaining proper module disabling

Commit:
fix(ci): restore configuration to match reference version

- Move upmeter back to disabled-modules.yaml (as in reference)
- Remove upmeter from mc.yaml to match reference structure
- Fix virtualization.yaml to use {{ .Values.virtualization.tag }} variable
- This should resolve upmeter timeout issues by matching working reference

Commit:
fix(ci): switch from CE to OSS registry with new credentials

- Replace deckhouse/modules-actions/setup@v2 with docker/login-action@v3
- Use BOOTSTRAP_DEV_REGISTRY_LOGIN/PASSWORD instead of DEV_MODULES_REGISTRY_*
- Switch dhctl image from CE to OSS: dev-registry.deckhouse.io/sys/deckhouse-oss/install:main
- This should resolve module timeout issues by using OSS registry

Commit:
fix(ci): restore token creation logic for nested kubeconfig

- Add ServiceAccount and ClusterRoleBinding creation for e2e-admin
- Add token generation via kubectl create token
- Use proper token in nested kubeconfig instead of empty string
- This should resolve EOF errors when accessing nested cluster

Commit:
feat(ci): add SDS module configuration for nested cluster

- Add ModuleConfig for sds-replicated-volume
- Add LVMVolumeGroup for data storage
- Add ReplicatedStoragePool with replication factor 2
- Add ReplicatedStorageClass linstor-thin-r2
- This should resolve SDS StorageClass not found errors

Commit:
feat(ci): add disk debugging and Ceph cluster configuration

- Add debug step to show disk information in nested cluster
- Add CephCluster configuration with nvme device filter
- Remove placement and tolerations sections as requested
- Set deviceFilter to ^nvme0n[1-9] for nvme disks

Commit:
refactor(ci): remove old Ceph bootstrap approach

- Remove ceph-bootstrap.sh script
- Remove nested:storage:ceph task
- Update nested:storage:configure to wait for Ceph StorageClass instead of bootstrap
- Ceph configuration now handled via Helm template (ceph-cluster.yaml)
- SDS configuration remains unchanged

Commit:
fix(ci): optimize Git checkout to resolve cache issues

- Remove fetch-depth: 0 from Setup Environment, Report Results, and Cleanup jobs
- Keep fetch-depth: 0 only in E2E jobs where full history is needed
- This should resolve Go cache and sha256sum verification issues
- Prevents cache corruption from multiple full Git checkouts

Commit:
fix(ci): install d8 via direct tarball to avoid checksum mismatch

- Replace d8-install.sh with direct download of d8 tar.gz
- Pin version to v0.9.3 and install binary to /usr/local/bin
- Eliminates failing sha256sum step in installer script

Commit:
fix(ci): install d8 via installer with checksum disabled

- Use official installer script again
- Set D8_SKIP_CHECKSUM=1 to bypass checksum verification
- Avoids tar layout issues and checksum mismatch failures

Commit:
feat(ci): conditionally render Ceph/SDS by storageProfile

- Wrap ceph-cluster.yaml with storageProfile==cephrbd
- Wrap sds.yaml with storageProfile!=cephrbd
- Set storageProfile in run values from matrix.profile in workflow
- Avoids unintended Rook/Ceph rendering for SDS profile

Commit:
refactor(ci): move storage CRs post-bootstrap and remove empty templates

- Remove ceph-cluster.yaml and sds.yaml from cluster-config templates
- Add flat manifests under ci/dvp-e2e/manifests/storage
- Apply storage backend after bootstrap in workflow
- Avoid bootstrap hangs and Rook/SDS races

Commit:
fix(ci): move disk debug before storage apply; add retries and --validate=false

- Run disk debug right after nested kubeconfig to aid Ceph deviceFilter
- Apply storage manifests with retries and without OpenAPI validation to avoid early 502
- Keeps configure storage classes step as-is

Commit:
fix(ci): move disk debug step before storage configuration

Commit:
fix(ci): use d8 v ssh for reliable VM disk debugging with retries

Commit:
fix(ci): correct Ceph deviceFilter and add proper operator readiness checks

- Fix deviceFilter from nvme to sd[bc] based on actual disk naming
- Add proper waiting for operator-ceph module readiness before applying CephCluster
- Set linstor-thin-r2 as default StorageClass for SDS profile
- Remove duplicate ModuleConfig from ceph.yaml

Commit:
fix(ci): make GA YAML valid; move operator-ceph ModuleConfig to file\n\n- Replace heredoc in workflow with file apply to avoid YAML parse errors\n- Add ci/dvp-e2e/manifests/storage/operator-ceph.yaml\n- Keep SDS default SC patch and Ceph readiness waits intact

Commit:
chore(ci): include Loop report scripts; add run link in profile cell\n\n- Restore loop_notify.py and loop_junit_notify.py from backup\n- Enhance loop_matrix_summary.py to embed GitHub run link on profile name\n- Keep single final Loop report without extra notifications

Commit:
chore(ci): pass GitHub run URL to Loop matrix summary and link profile name

Commit:
fix(storage): enable SDS modules and wait for CRDs before applying resources\n\n- Workflow: enable sds-node-configurator and sds-replicated-volume, wait for CRDs, then apply sds.yaml\n- Taskfile: mirror SDS enable+CRD wait in nested:storage:sds\n- Keeps Ceph path intact with retries

Commit:
fix(ci): resolve YAML syntax error in workflow SDS module config\n\n- Replace heredoc with separate sds-modules.yaml file\n- Fixes 'Unexpected value apiVersion' error in GitHub Actions\n- YAML syntax now valid

Commit:
fix(storage): use kubectl patch mc global for default StorageClass\n\n- Replace forbidden kubectl patch storageclass with kubectl patch mc global\n- Use JSON patch to set defaultClusterStorageClass in ModuleConfig global\n- Add fallback selector for Ceph rook-ceph-operator pod\n- Apply to both SDS (linstor-thin-r2) and Ceph (ceph-pool-r2-csi-rbd-immediate)

Commit:
fix(sds): add retry logic and remove version from ModuleConfig\n\n- Add retry loop for sds-modules.yaml apply to handle 502 errors\n- Remove spec.version from ModuleConfig (no effect without settings)\n- Use --validate=false and --request-timeout=10s for robustness

Commit:
feat(ci): improve SDS CRD waiting and Ceph operator readiness checks\n\n- Add two-stage SDS CRD waiting: first check existence, then wait for Established\n- Increase Ceph namespace timeout from 30 to 60 attempts (5→10 minutes)\n- Add fail-fast check for d8-operator-ceph namespace creation\n- Increase Ceph pod wait timeouts: 60s→120s, 240s→480s\n- Better resilience for slow runners and temporary API server issues

Commit:
fix(storage): correct disk devices and simplify SDS logic

- Fix SDS disk: /dev/sdb → /dev/sdd in LVMVolumeGroup
- Fix Ceph disk: deviceFilter "^sd[bc]$" → "^sdd$"
- Simplify SDS: remove complex module enabling, keep simple sds.yaml apply
- Improve Ceph: increase timeouts, add fail-fast checks, fallback selectors
- Use kubectl patch mc global for default StorageClass (both SDS and Ceph)
- Add retry logic and better error handling throughout
- Add infra:attach-worker-disks task for automatic data disk attachment
- Update profiles.json with worker_data_disk_size configuration
- Update get_profile_config.sh to export ATTACH_DISK_SIZE variable

Commit:
fix(scripts): fix jq syntax error in get_profile_config.sh

- Remove escaped quotes in jq expression for worker_data_disk_size
- Fix: "10Gi" → "10Gi" in jq string interpolation

Commit:
fix(taskfile): fix heredoc syntax in infra:attach-worker-disks task

- Replace heredoc with cat > /tmp files approach to avoid YAML parsing issues
- EOF must be unindented in heredoc, which conflicts with YAML structure
- Use temporary files for VirtualDisk and VirtualMachineBlockDeviceAttachment manifests

Commit:
fix(taskfile): avoid 'workers: unbound variable' by safe array build

- Initialize workers=()
- Read kubectl output via while-read, ignore empty lines
- Print number of discovered workers

Commit:
fix(taskfile): add quotes around SDS_SC_NAME in storageclass check

- Prevents bash syntax errors if storage class name contains spaces/special chars

Commit:
fix(ci): enable JUnit XML generation and reporting in E2E matrix workflow

- Fix JUNIT_PATH to use absolute path instead of relative
- Add JUnit directory preparation step before test execution
- Pass JUNIT_PATH variable through Taskfile nested:e2e task
- Add --junit-report support to ginkgo command in tests/e2e/Taskfile.yaml
- Add kubectl configuration to report job for Secret creation
- Add verification step to check JUnit file creation
- Enhance debug information in report job

This fixes the issue where JUnit reports were not being generated,
causing the loop_matrix_summary.py script to fail with 'No results to report'.

Commit:
fix(taskfile): correct worker VM selector in infra:attach-worker-disks

- Change from dvp.deckhouse.io/node-group=worker to hostname-based selection
- Use jq to filter VMs with hostname containing 'worker'
- Fixes issue where worker VMs exist but have different label structure

Commit:
fix: apply critical storage and disk attachment fixes

- Fix VirtualDisk API structure (use size + storageClassName instead of resources/accessModes)
- Change SDS device from /dev/sdd to /dev/vdb
- Fix TARGET_STORAGE_CLASS for Ceph bootstrap (use linstor-thin-r2 instead of ceph SC)
- Fix Secret name generation to lowercase only
- Revert worker VM selector to use node-group label
- Note: Taskfile YAML syntax needs final heredoc fix

Commit:
hotfix: fix Taskfile YAML syntax and VirtualDisk API structure

- Fix VirtualDisk to use size + storageClassName (not resources/accessModes)
- Revert worker VM selector to use node-group label
- Fix YAML syntax issues with heredoc

Commit:
feat: move disk attachment before bootstrap and use base SC

- Move 'Attach data disks to worker VMs' before 'Bootstrap nested cluster'
- Use base StorageClass 'linstor-thin-r2' for disk attachment (available in parent cluster)
- This prevents bootstrap hanging on non-existent Ceph StorageClass
- Disks are attached early, storage backend configured later

Commit:
fix: use VM name selector instead of node-group label for worker detection

- Worker VMs don't have dvp.deckhouse.io/node-group=worker label
- Use grep worker on VM names instead of label selector
- This fixes 'No worker VMs found' error in attach-worker-disks task

Commit:
fix: move disk attachment after bootstrap to ensure VMs exist

- Move 'Attach data disks to worker VMs' after 'Bootstrap nested cluster'
- Worker VMs are created during bootstrap, not before
- This fixes 'No worker VMs found' error when trying to attach disks too early
- Maintains base StorageClass 'linstor-thin-r2' for disk attachment

Commit:
fix(ci): resolve SDS and Ceph storage configuration issues

- Add multiple disk attachment (2 disks per VM) to avoid conflicts with system disks
- Add missing sds-local-volume module for SDS CRD creation
- Fix device mapping: SDS uses /dev/sdd, Ceph uses /dev/sde
- Add missing Ceph modules: csi-ceph and snapshot-controller
- Improve CephCluster configuration with health checks
- Update workflow to use DISK_COUNT parameter

Fixes issues with missing CRDs and namespace creation timeouts.

Commit:
fix(ci): add SCSI bus rescan to activate hotplugged storage disks

- Add SCSI bus rescan command to master VM debug step
- Add separate step to activate storage disks on all worker VMs
- This should make hotplugged disks visible in lsblk output
- Fixes issue where disks are attached but not visible in guest OS

Commit:
fix(ci): add sudo to SCSI rescan commands to fix permission denied errors

Commit:
feat(ci): implement blockDeviceRefs approach for VM disk attachment

- Replace hotplug with blockDeviceRefs to avoid ExpandDisks featuregate issues
- Add new task infra:attach-storage-disks-via-blockdevicerefs
- Remove SCSI rescan steps as they are no longer needed
- Clean up comments and deprecated messages

Commit:
fix(ci): improve blockDeviceRefs task reliability

- Remove namespace from VirtualDisk metadata (not needed)
- Use JSON file for patch instead of inline JSON to avoid escaping issues
- Add proper VM state checking before stopping
- Add error handling for VM stop failures
- Improve logging for better debugging

Commit:
fix(ci): use correct VM selector for worker nodes

- Replace label selector dvp.deckhouse.io/node-group=worker with name-based grep
- Worker VMs don't have the expected label, causing task to find 0 VMs
- This fixes blockDeviceRefs attachment for storage disks

Commit:
fix(ci): add debug info and validation for blockDeviceRefs patch

- Add debug output to show block_device_refs content
- Add validation to skip VMs with empty blockDeviceRefs
- Add patch file content logging for troubleshooting
- This should fix the 'must specify --patch' error

Commit:
fix(ci): use inline patch instead of file-based patch for blockDeviceRefs

- Replace file-based kubectl patch with inline patch
- Remove temporary file creation and heredoc issues
- Use escaped JSON in kubectl patch -p parameter
- This should fix the 'must specify --patch' error completely

Commit:
fix(ci): restore correct namespace naming for cleanup

- Change RUN_ID from 'e2e-' to 'nightly-nested-e2e-' prefix
- This ensures namespace cleanup can find and delete test namespaces
- Namespaces will now be created as 'nightly-nested-e2e-sds-XXXX' and 'nightly-nested-e2e-cephrbd-XXXX'
- Matches the FILTER_PREFIX='nightly-nested-e2e-' used in cleanup job

Commit:
fix(ci): add VM startup verification after blockDeviceRefs modification

- Add verification that VM successfully starts after adding blockDeviceRefs
- Wait up to 10 minutes for VM to reach Running state
- Fail fast if VM fails to start after blockDeviceRefs modification
- Prevents subsequent workflow steps from running on non-functional VMs

Commit:
revert(ci): switch back to hotplug for disk attachment

- Replace blockDeviceRefs approach with VirtualMachineBlockDeviceAttachment
- Hotplug attaches disks to running VMs without stopping them
- This should resolve VM crashes during disk attachment stage

Commit:
fix(ci): fix E2E test issues with bootstrap and disk attachment

- Add worker nodes configuration in cluster-config.yaml
- Fix storage class selection logic in workflow
- Add namespace to VirtualDisk and VMBDA manifests
- Remove excessive debug logs and checks
- Fix YAML syntax in storage manifests
- Use ceph-pool-r2-csi-rbd-immediate for Ceph profile

Fixes: bootstrap and disk attachment failures in E2E tests

Commit:
fix(ci): remove forbidden dataDisk property from DVPClusterConfiguration

- Remove dataDisk from nodeGroups instanceClass (not supported in DVPClusterConfiguration)
- Fix YAML structure with proper indentation for nodeGroups
- Remove unused dataDiskSize and data fields from values.yaml
- Data disks will be attached via hotplug mechanism instead

Fixes: bootstrap error 'dataDisk is a forbidden property'

Commit:
fix: remove nodeGroups from DVPClusterConfiguration

- Remove nodeGroups section from DVPClusterConfiguration as it's not supported
- Worker nodes are created via separate DVPInstanceClass and NodeGroup resources
- This matches the architecture from backup branch where only masterNodeGroup is defined
- Fixes bootstrap error: nodeGroups.instanceClass.dataDisk is a forbidden property

Commit:
docs: add comment to cluster-config template

- Add descriptive comment to cluster-config.yaml template
- Clarify purpose of the configuration file for DVP-over-DVP E2E testing

Commit:
fix(ci): add push trigger to E2E Matrix Tests workflow

- Add push event trigger for main and feat/ci-e2e-matrix branches
- Workflow was only running on PR events, schedule, and manual dispatch
- Now it will also trigger on direct pushes to these branches

Commit:
fix(ci): limit push trigger to feat/ci-e2e-matrix branch only

- Remove main branch from push trigger to avoid unnecessary runs
- Keep pull_request trigger for both main and feat/ci-e2e-matrix
- Now workflow will only trigger on push to our feature branch

Commit:
refactor(ci): simplify storage configuration logic

- Always use ceph-pool-r2-csi-rbd-immediate as default storage class
- Remove conditional logic for storage profile selection
- Always enable Ceph as base storage backend for all profiles
- Enable SDS modules only for 'sds' profile as additional layer
- Simplify bootstrap and disk attachment storage class configuration
- Reduce complexity and improve reliability of E2E tests

Commit:
fix(ci): update storage configuration to use Ceph as default

- Set defaultClusterStorageClass to ceph-pool-r2-csi-rbd-immediate in mc.yaml
- Improve Ceph deviceFilter to support various disk types (sd, vd, xvd)
- Exclude system disks (sda/sdb/sdc) from Ceph OSD usage
- Align with simplified storage logic using Ceph everywhere

Commit:
debug(ci): improve disk attachment debugging and reduce retry time

- Reduce retry attempts from 120 to 30 (2.5 minutes instead of 10 minutes)
- Add comprehensive debugging for disk attachment failures:
  * PVC status and describe output
  * StorageClass availability check
  * Ceph CSI pods status
  * Ceph cluster status
  * Enhanced periodic debug snapshots every 10 retries
- Improve error reporting with detailed resource descriptions
- Add debugging to both disk attachment tasks for consistency

Commit:
debug(ci): improve PVC waiting logic and add debug environment variable

Taskfile.yaml changes:
- Replace kubectl wait with custom polling loop for better control
- Add detailed PVC phase tracking with retry counter
- Extract and display StorageClass, PV name, and VolumeMode separately
- Add comprehensive error reporting with describe and events output
- Increase PVC wait timeout from 300s to 240s (120 retries * 2s)

Workflow changes:
- Add DEBUG_HOTPLUG environment variable for enhanced debugging output

Commit:
debug(ci): optimize logging and add fallback for disk attachment

- Reduce log verbosity: print status every 30s instead of every 5s
- Reduce debug snapshots frequency: every 60s instead of every 50s
- Add fallback mechanism: check VM events for successful hotplug completion
- Filter controller/handler logs by namespace/VM/VD to reduce noise
- Improve error reporting by focusing on relevant logs only
- This helps diagnose race conditions where disk is attached but VMBDA status not updated

Commit:
feat: add ModulePullOverride for Ceph and SDS modules

- Add MRO for operator-ceph, csi-ceph, snapshot-controller modules
- Add MRO for sds-node-configurator, sds-local-volume, sds-replicated-volume modules
- Use main tag and 15s scan interval for dev environment

Commit:
fix: improve disk attachment fallback logic and use dynamic storage class

- Add Fallback A: check VM status for attached/hotplugged disk before falling back to events
- Reorder fallback logic to check VM status first, then VM events
- Use dynamic STORAGE_CLASS from profile config instead of hardcoded value
- Improve error handling and debug output structure

Commit:
refactor: simplify disk attachment logic and remove debug env

- Remove DEBUG_HOTPLUG environment variable from workflow
- Simplify fallback logic: check VM status earlier in polling loop
- Reduce periodic debug output to minimal summary
- Early success detection using VM blockDeviceRefs status

Commit:
refactor: compactify jq query formatting in disk attachment logic

Commit:
revert: use hardcoded ceph storage class for disk attachment

- Revert to explicit ceph-pool-r2-csi-rbd-immediate instead of dynamic STORAGE_CLASS
- This ensures consistent behavior during hotplug attachment phase

Commit:
improve: add Deckhouse readiness check and better error handling

- Add explicit Deckhouse deployment rollout check before enabling modules
- Wait for Deckhouse service before applying ModuleConfigs
- Increase timeouts for namespace creation (60s -> 90s) and pod waits (120s -> 300s)
- Increase retry attempts for module applies (10 -> 12)
- Add detailed error diagnostics when namespace creation fails
- Improve error messages with filtered Deckhouse logs on failure

Commit:
feat(ci): add platform queue diagnostics for module deployment debugging

- Add queue list output in nested_diag.sh deep mode
- Add queue diagnostics in e2e-matrix.yml after Deckhouse rollout
- Add queue diagnostics on namespace creation failure
- Queue shows module deployment errors and task states

Commit:
fix(ci): improve queue diagnostics error handling and output

- Add detailed error messages for queue command failures
- Redirect stderr to stdout to see actual errors
- Check if d8 command exists before using it
- Add intermediate debug messages for troubleshooting

Commit:
fix(ci): replace d8 v ssh queue diagnostics with kubectl-based debugging

d8 CLI is not available inside nested cluster. Replace queue diagnostics with:
- Deckhouse deployment status and conditions
- Recent Deckhouse events
- ModuleConfig status for operator-ceph and csi-ceph
- ModuleConfig-related events

This provides better visibility into module deployment issues without requiring d8 CLI in nested cluster.

Commit:
feat(ci): execute d8 platform queue command via kubectl exec in deckhouse pod

Use kubectl exec to access d8 platform queue list from inside deckhouse pod, which already has d8 CLI available. This is simpler than installing d8 CLI separately in nested cluster.

- Get deckhouse pod name using nested kubeconfig
- Execute d8 platform queue list command via kubectl exec
- Add queue diagnostics both after successful Deckhouse rollout and on namespace creation failure

Commit:
feat(scripts): add automatic d8 CLI installation on master VM for diagnostics

- Check if d8 CLI exists on master VM before running queue diagnostics
- Install d8 CLI automatically if missing using official install script
- Filter out SSH authenticity warnings from output
- This enables queue diagnostics in nested clusters when d8 CLI is not pre-installed

Commit:
feat(ci): install d8 CLI in deckhouse pod before queue diagnostics

- Check if d8 CLI exists in deckhouse pod container
- Install d8 CLI using official install script if missing
- Execute d8 platform queue list after installation
- This ensures queue diagnostics work reliably without requiring d8 CLI in container image

Commit:
fix(ci): install tar and use official d8 CLI installation method

- Install tar package via apt-get for Ubuntu-based deckhouse container
- Use official Deckhouse CLI installation method from documentation
- Download and extract d8 binary from official tar.gz archive
- This ensures reliable d8 CLI installation without relying on install script

Commit:
fix(ci): support both Alpine and Ubuntu for d8 CLI installation

- Detect distro using apk (Alpine) or apt-get (Ubuntu)
- Install tar and curl packages based on distro
- Download to /tmp directory to avoid read-only filesystem issues
- Install d8 CLI to /usr/local/bin for both distros

Commit:
ci: use runner d8 for nested (e2e-matrix.yml)

Commit:
ci: stabilize d8 queue + webhook wait (e2e-matrix.yml)

Commit:
ci: persist SSH key secret (Taskfile.yaml)

Commit:
scripts: nested diag: auto-fetch SSH key + ssh opts (nested_diag.sh)

Commit:
ci: pin d8 v0.13.2 on runner (e2e-matrix.yml)

Commit:
scripts: nested_diag.sh: ensure queue-capable d8 inside VM

Commit:
scripts: nested_diag.sh: use admin.conf for d8 queue

Commit:
scripts: nested_diag.sh: fallback to kubectl config view for kubeconfig

Commit:
scripts: nested_diag.sh: robust kubeconfig for d8 (avoid 127.0.0.1:6445, use SA token)

Commit:
ci(e2e-matrix): fix Ceph and storage setup; import GH SSH keys

- Taskfile: add ssh:import-gh and ssh:ensure; use in render-infra
- Workflow: render-infra/infra-deploy use USE_GH_SSH_KEYS=true
- Workflow: for cephrbd, enable Ceph modules and apply CephCluster
- Keep SDS config in dedicated Task step, remove invalid sds-modules ref

Commit:
ci(e2e-matrix): remove accidental embedded repo from index

Commit:
ci(e2e-matrix): pin TARGET_STORAGE_CLASS to Ceph immediate SC

Use explicit 'ceph-pool-r2-csi-rbd-immediate' for nested storage configuration.

Commit:
ci(e2e-matrix): drop accidentally staged embedded repo

Commit:
ci(e2e-matrix): default storage profile to ceph and guard empty matrix.profile

- Taskfile: default STORAGE_PROFILE=ceph in nested:storage:configure
- Workflow: fail fast if matrix.profile is empty before storage config

Commit:
ci(e2e-matrix): drop accidentally staged embedded repo

Commit:
chore(ci): disable virtualization in e2e matrix

Commit:
ci(e2e): Ceph in nested cluster via deckhouse-prod ModuleSource; enable Ceph modules; wait Ready; add CSI smoke; use parent Ceph SC for hotplug

Commit:
ci(e2e): re-enable virtualization in nested cluster

Commit:
ci(e2e): use NFS as initial default SC for nested; add in-cluster NFS server; enable csi-nfs; keep Ceph switch post-bootstrap

Commit:
Revert "ci(e2e): use NFS as initial default SC for nested; add in-cluster NFS server; enable csi-nfs; keep Ceph switch post-bootstrap"

This reverts commit 131189e0e49024082c72769092cae03f1da2e93c.

Commit:
Configure Ceph to use CephStorageClass instead of direct StorageClass

- Add CephClusterConnection creation after CephCluster is ready
- Disable cloud-provider-dvp module to avoid StorageClass conflicts
- Replace direct StorageClass with CephStorageClass CRD
- Remove StorageClass from ceph.yaml manifest
- Use correct provisioner rbd.csi.ceph.com instead of rook-ceph.rbd.csi.ceph.com

Commit:
Fix sds-replicated-volume configuration

- Add type: LVMThin to ReplicatedStoragePool
- Update lvmVolumeGroups to use name and thinPoolName objects
- Set correct API version for ReplicatedStoragePool and ReplicatedStorageClass
- Add volumeAccess: Local to ReplicatedStorageClass

Commit:
Remove redundant StorageClass check for Ceph in nested:storage:configure

StorageClass is already validated in workflow with correct provisioner check,
so the duplicate check in Taskfile is redundant.

Commit:
Fix heredoc syntax error in CephClusterConnection creation

- Move command substitution outside heredoc to avoid syntax issues
- Add validation for MONITORS_LIST to catch errors early

Commit:
Fix yamllint issues: remove trailing spaces and fix colon spacing

- Remove all trailing spaces from workflow and Taskfile
- Fix too many spaces after colon in Taskfile.yaml line 629

Commit:
Remove incorrect source: deckhouse-prod from virtualization ModuleConfig

virtualization module should use default deckhouse source, not deckhouse-prod.
deckhouse-prod is only needed for Ceph modules (operator-ceph, csi-ceph).

Commit:
fix(ci): add retry for token creation and fix heredoc syntax in workflow

- Add retry loop for create token command in nested:kubeconfig task to handle periodic failures
- Fix heredoc syntax error by generating MONITORS_YAML before heredoc block instead of using command substitution inside

Commit:
fix(ci): replace heredoc with echo block to fix YAML syntax error

GitHub Actions YAML parser cannot handle multiline variable substitution inside heredoc.
Replace heredoc with echo block to generate CephClusterConnection manifest safely.

Commit:
chore(ci): remove nested_diag.sh from matrix branch (unused)

Commit:
ci: tidy e2e matrix workflow

Commit:
ci: stabilize heredoc in e2e matrix

Commit:
ci: drop heredocs from e2e matrix

Commit:
ci: generate manifests with yq in e2e matrix

Commit:
ci: trim redundant waits in e2e matrix

Commit:
ci: wire prod registry and fix sds crd wait

Commit:
ci: keep dev registry login for dhctl bootstrap

Commit:
ci: include dev registry in dockercfg

Commit:
ci: stabilize nested kubeconfig and sds setup

Commit:
chore(ci): add ingress publish smoke to nested e2e matrix (ingress-nginx HostPort, in-cluster curl)

Commit:
ci: harden module apply and nested ssh retries

Commit:
ci: simplify cleanup workflow

Commit:
chore(ci): remove leftover comment from nested e2e matrix flow

Commit:
ci: fix sds module heredoc

Commit:
chore(ci): split e2e preparation and test jobs

Commit:
ci(e2e): fix heredoc EOFs; route Ceph setup via Taskfile and add waits for CRDs/cluster connection

Commit:
ci: fix YAML by reindenting heredoc EOF markers

Commit:
ci(e2e): route storage selection via shell case; drop ingress smoke steps

Commit:
ci(e2e): fix SDS ModuleConfig apply to avoid heredoc; generate YAML via yq

Commit:
ci(e2e): fix relative paths for ceph manifests; add cwd debug; remove early Go setup in prepare

Commit:
ci(e2e): stop polling Deckhouse Module states; wait for CRDs/operator instead (SDS/Ceph)

Commit:
ci(e2e): deflake SA token retrieval — wait for SA + fallback to token Secret if TokenRequest fails

Commit:
ci(e2e): use parent token from GitHub secret for bootstrap; prefer kubeconfig token; keep TokenRequest as last resort

Commit:
ci(e2e): revert kubeconfig token flow to TokenRequest to avoid breaking bootstrap; remove PARENT_SA_TOKEN wiring

Commit:
ci(e2e): ensure ModuleSource for SDS; set ModuleConfig.source=deckhouse-prod; apply SDS manifests; use profile storage class

Commit:
ci(sds): fix LVMVolumeGroup apiVersion to storage.deckhouse.io/v1alpha1

Commit:
ci: trim noisy debug in Ceph task; fix SDS LVMVolumeGroup spec to Local with selector

Commit:
ci(sds): fix LVMVolumeGroup schema (spec.local) and use LVM pool; ci(ceph): wait for operator-ceph OR rook-ceph-operator

Commit:
ci(ceph): wait only for deploy/operator-ceph (drop rook-ceph-operator fallback)

Commit:
ci: normalize waits to 50 iterations (was 36/60) and update log counters

Commit:
ci(sds): fix module source conflict (use Embedded) and yq env var for per-node LVMVG generation

Commit:
ci(sds): use deckhouse-prod (io) via REGISTRY_DOCKER_CFG; fix yq env(NODE) in per-node LVMVG generation

Commit:
ci(e2e-matrix): use profile SC; drop aliases; fix summary; stable RUN_ID (no SHA); no docker prune

Commit:
ci(e2e-matrix): keep Ceph SC for bootstrap and worker hotplug as base fast storage
